core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -130,63 +126,105 @@ inline void check_shmem_request(CudaInternal const* cuda_instance, int shmem) {
   }
 }
 
-// This function needs to be template on DriverType and LaunchBounds
+// These functions need to be templated on DriverType and LaunchBounds
 // so that the static bool is unique for each type combo
 // KernelFuncPtr does not necessarily contain that type information.
+
+template <class DriverType, class LaunchBounds, class KernelFuncPtr>
+const cudaFuncAttributes& get_cuda_kernel_func_attributes(
+    const KernelFuncPtr& func) {
+  // Only call cudaFuncGetAttributes once for each unique kernel
+  // by leveraging static variable initialization rules
+  auto wrap_get_attributes = [func]() -> cudaFuncAttributes {
+    cudaFuncAttributes attr_tmp;
+    KOKKOS_IMPL_CUDA_SAFE_CALL(cudaFuncGetAttributes(&attr_tmp, func));
+    return attr_tmp;
```

Comment:
```suggestion
    cudaFuncAttributes attr;
    KOKKOS_IMPL_CUDA_SAFE_CALL(cudaFuncGetAttributes(&attr, func));
    return attr;
```
You are typing more and the `_tmp` suffix does not improve readability.

Reply:
changed